community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/SchemaProcedure.java

```diff
@@ -345,20 +368,43 @@ public String toString()
         {
             return String.format( "VirtualRelationshipHack[%s]", id );
         }
+
+        @Override
+        public NodeValue startNode()
+        {
+            return null;
+        }
+
+        @Override
+        public NodeValue endNode()
+        {
+            return null;
+        }
+
+        @Override
+        public TextValue type()
+        {
+            return null;
+        }
+
+        @Override
+        public MapValue properties()
+        {
+            return null;
```

Comment:
If we expect this to not be invoked, throwing an exception is much better than returning a value that can wreck havoc in other places.